### PR TITLE
BUG: Allow spaces in output string of einsum

### PR DESCRIPTION
**Pull request by eric-wieser**
Also produce more useful error messages

Fixes gh-10794

### CODE REVIEW

File: numpy/core/tests/test_einsum.py [hunk 1]
```diff
@@ -918,6 +920,13 @@ def test_path_type_input(self):
         opt = np.einsum(*path_test, optimize=exp_path)
         assert_almost_equal(noopt, opt)
 
+    def test_spaces(self):
+        #gh-10794
+        arr = np.array([[1]])
+        for sp in itertools.product(['', ' '], repeat=4):
+            # no error for any spacing
+            np.einsum('{}...a{}->{}...a{}'.format(sp), arr)
```
Comment (mhvk) [inline numpy/core/tests/test_einsum.py]: Should either have `{0}` for all, or maybe `{sp}...'.format(sp=sp)`

**Review by mhvk — COMMENTED** [review]
Apart from the mistake in the test case string formatting, this looks good.

File: numpy/core/tests/test_einsum.py (hunk 1 above)
Comment (eric-wieser) [reply numpy/core/tests/test_einsum.py]: Meant to use a * here

**Comment by eric-wieser** [comment]
@mhvk: Fixed

**Review by mhvk — APPROVED** [review]
Looks all OK now. Will merge.

**Pull request merged**